### PR TITLE
Add rate limiter to limit disk IO when doing compaction and flush

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
@siddontang @BusyJay PTAL

### CODE REVIEW

File: src/rocksdb_options.rs [hunk 1]
```diff
@@ -603,7 +624,17 @@ impl Options {
...
         unsafe {
-            crocksdb_ffi::crocksdb_options_set_compaction_readahead_size(self.inner, size as size_t);
+            crocksdb_ffi::crocksdb_options_set_compaction_readahead_size(self.inner,
+                                                                         size as size_t);
+        }
+    }
+
+    pub fn set_ratelimiter(&mut self, rate_bytes_per_sec: i64) {
+        let rate_limiter = RateLimiter::new(rate_bytes_per_sec,
+                                            100 * 1000, // 100ms should work for most cases
```
Comment (siddontang) [inline src/rocksdb_options.rs]: use a const var

**Comment by siddontang** [comment]
Please fix conflicts first.

File: src/rocksdb_options.rs [hunk 2]
```diff
@@ -95,6 +95,30 @@ impl BlockBasedOptions {
...
+                                                      refill_period_us,
+                                                      fairness)
+        };
+        RateLimiter { inner: limiter }
+    }
+}
+
+impl Drop for RateLimiter {
+    fn drop(&mut self) {
+        unsafe { crocksdb_ffi::crocksdb_ratelimiter_destroy(self.inner) }
```
Comment (siddontang) [inline src/rocksdb_options.rs]: safe to destroy here?

Comment (zhangjinpeng1987) [inline src/rocksdb_options.rs]: yes, just free the wrapper.

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @BusyJay

**Comment by BusyJay** [comment]
LGTM

**Comment by siddontang** [comment]
Please fix the conflict @zhangjinpeng1987

**Comment by siddontang** [comment]
Ping @zhangjinpeng1987

**Pull request merged**